core/src/services/icloud/drive.rs

```diff
@@ -0,0 +1,201 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use http::header::{IF_MATCH, IF_NONE_MATCH};
+use http::{Method, Request, Response, StatusCode};
+use log::debug;
+use serde_json::json;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+use crate::raw::oio::WriteBuf;
+use crate::raw::{new_json_deserialize_error, AsyncBody, IncomingAsyncBody, OpRead};
+use crate::Result;
+
+use super::core::{parse_error, IcloudItem, IcloudObject, IcloudRoot, IcloudSigner};
+
+#[derive(Clone)]
+pub struct File {
+    pub id: Option<String>,
+    pub name: String,
+    pub size: u64,
+    pub date_created: Option<String>,
+    pub date_modified: Option<String>,
+    pub mime_type: String,
+}
+
+// A directory in icloud Drive.
+#[derive(Clone)]
+pub struct Folder {
+    pub id: Option<String>,
+    pub name: String,
+    pub date_created: Option<String>,
+    pub items: Vec<IcloudItem>,
+    pub mime_type: String,
+}
+
+pub struct FolderIter<'a> {
+    current: std::slice::Iter<'a, IcloudItem>,
+}
+
+impl<'a> Iterator for FolderIter<'a> {
+    type Item = &'a IcloudItem;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.current.next()
+    }
+}
+
+// A node within the icloud Drive filesystem.
+#[derive(Clone)]
+pub enum DriveNode {
```

Comment:
Seems we are not using this part of code. How about removing them all in this PR?

Reply:
done